2024-07-11 12:35:51.052739: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:35:51.053467: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:35:59,693 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:35:59,788 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:35:59,788 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:35:59,789 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:36:00,959 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:36:00,959 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:00,975 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4066666666666667}
INFO flwr 2024-07-11 12:36:00,975 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:36:01,167 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:01,194 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:36:01,196 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,204 | server.py:131 | fit progress: (1, 0.41732801589234314, {'accuracy': 0.82}, 0.22873590000000021)
DEBUG flwr 2024-07-11 12:36:01,204 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:01,224 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:36:01,224 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:36:01,225 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:01,250 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,261 | server.py:131 | fit progress: (2, 0.41465029099405365, {'accuracy': 0.8166666666666667}, 0.2850507000000011)
DEBUG flwr 2024-07-11 12:36:01,261 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:36:01,278 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,279 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,305 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,318 | server.py:131 | fit progress: (3, 0.41367338091694944, {'accuracy': 0.8166666666666667}, 0.3416592999999999)
DEBUG flwr 2024-07-11 12:36:01,318 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,337 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,338 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,363 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,374 | server.py:131 | fit progress: (4, 0.41367661105140274, {'accuracy': 0.8166666666666667}, 0.3981401000000009)
DEBUG flwr 2024-07-11 12:36:01,375 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,396 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,396 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,424 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,434 | server.py:131 | fit progress: (5, 0.41367662900575813, {'accuracy': 0.8166666666666667}, 0.45884760000000036)
DEBUG flwr 2024-07-11 12:36:01,434 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,455 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,455 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:36:01,480 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,495 | server.py:131 | fit progress: (6, 0.41367662943759304, {'accuracy': 0.8166666666666667}, 0.519039900000001)
DEBUG flwr 2024-07-11 12:36:01,495 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:01,529 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,530 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:36:01,566 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,580 | server.py:131 | fit progress: (7, 0.4125934228292241, {'accuracy': 0.8133333333333334}, 0.6043416000000015)
DEBUG flwr 2024-07-11 12:36:01,581 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,607 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,608 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,643 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,655 | server.py:131 | fit progress: (8, 0.41267141113889677, {'accuracy': 0.8066666666666666}, 0.6798691000000012)
DEBUG flwr 2024-07-11 12:36:01,655 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,684 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,685 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,728 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,741 | server.py:131 | fit progress: (9, 0.4126707503206595, {'accuracy': 0.8066666666666666}, 0.7648634000000012)
DEBUG flwr 2024-07-11 12:36:01,741 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,771 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:36:01,771 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,814 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:36:01,829 | server.py:131 | fit progress: (10, 0.41267076975015077, {'accuracy': 0.8066666666666666}, 0.8530364000000006)
DEBUG flwr 2024-07-11 12:36:01,829 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:36:01,857 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:36:01,857 | server.py:153 | FL finished in 0.8809064000000006
INFO flwr 2024-07-11 12:36:01,857 | app.py:225 | app_fit: losses_distributed [(1, 0.4001797301073869), (2, 0.4000503433247407), (3, 0.404893292983373), (4, 0.4048934062321981), (5, 0.4048934052387873), (6, 0.41425601315909416), (7, 0.41629829960209985), (8, 0.4161646851471492), (9, 0.4161646476813725), (10, 0.4161646447011403)]
INFO flwr 2024-07-11 12:36:01,858 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:36:01,858 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:36:01,858 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.41732801589234314), (2, 0.41465029099405365), (3, 0.41367338091694944), (4, 0.41367661105140274), (5, 0.41367662900575813), (6, 0.41367662943759304), (7, 0.4125934228292241), (8, 0.41267141113889677), (9, 0.4126707503206595), (10, 0.41267076975015077)]
INFO flwr 2024-07-11 12:36:01,858 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.4066666666666667), (1, 0.82), (2, 0.8166666666666667), (3, 0.8166666666666667), (4, 0.8166666666666667), (5, 0.8166666666666667), (6, 0.8166666666666667), (7, 0.8133333333333334), (8, 0.8066666666666666), (9, 0.8066666666666666), (10, 0.8066666666666666)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.08633463,  0.99806941,  0.69456748,  0.43285394,  0.18846479,
         0.06466852,  0.08849428,  0.06158617]]), array([0.05180895])]
[array([[-2.08633463,  0.99806941,  0.69456748,  0.43285394,  0.18846479,
         0.06466852,  0.08849428,  0.06158617]]), array([0.05180895])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05242987,  0.98462557,  0.71651329,  0.43615205,  0.17688084,
         0.05246469,  0.05778788,  0.05503928]]), array([0.07712086])]
[array([[-2.05242987,  0.98462557,  0.71651329,  0.43615205,  0.17688084,
         0.05246469,  0.05778788,  0.05503928]]), array([0.07712086])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01588248,  0.95346281,  0.75871256,  0.39646828,  0.18212055,
         0.04843563,  0.0943519 ,  0.04101536]]), array([0.08319103])]
[array([[-2.01588248,  0.95346281,  0.75871256,  0.39646828,  0.18212055,
         0.04843563,  0.0943519 ,  0.04101536]]), array([0.08319103])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01577911,  0.95357175,  0.75860436,  0.39639641,  0.18207792,
         0.04833119,  0.09441237,  0.04112847]]), array([0.08310116])]
[array([[-2.01577911,  0.95357175,  0.75860436,  0.39639641,  0.18207792,
         0.04833119,  0.09441237,  0.04112847]]), array([0.08310116])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01577782,  0.95357058,  0.75860336,  0.39639577,  0.18207776,
         0.04833083,  0.09441247,  0.04113042]]), array([0.08310074])]
[array([[-2.01577782,  0.95357058,  0.75860336,  0.39639577,  0.18207776,
         0.04833083,  0.09441247,  0.04113042]]), array([0.08310074])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01577782,  0.95357057,  0.75860335,  0.39639576,  0.18207777,
         0.04833083,  0.09441248,  0.04113043]]), array([0.08310074])]
[array([[-2.01577782,  0.95357057,  0.75860335,  0.39639576,  0.18207777,
         0.04833083,  0.09441248,  0.04113043]]), array([0.08310074])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9653922 ,  0.95675106,  0.75659042,  0.39272016,  0.17271061,
         0.07592529,  0.04962977,  0.0189395 ]]), array([0.06472603])]
[array([[-1.9653922 ,  0.95675106,  0.75659042,  0.39272016,  0.17271061,
         0.07592529,  0.04962977,  0.0189395 ]]), array([0.06472603])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96225827,  0.940746  ,  0.75228926,  0.39874875,  0.19573416,
         0.07674539,  0.02663711,  0.02959609]]), array([0.03916759])]
[array([[-1.96225827,  0.940746  ,  0.75228926,  0.39874875,  0.19573416,
         0.07674539,  0.02663711,  0.02959609]]), array([0.03916759])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96220962,  0.94073923,  0.75232206,  0.39874353,  0.19569715,
         0.07680544,  0.02665312,  0.0295585 ]]), array([0.03911039])]
[array([[-1.96220962,  0.94073923,  0.75232206,  0.39874353,  0.19569715,
         0.07680544,  0.02665312,  0.0295585 ]]), array([0.03911039])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9622107 ,  0.94073996,  0.75232206,  0.39874223,  0.19569684,
         0.07680474,  0.02665377,  0.02955768]]), array([0.03911377])]
[array([[-1.9622107 ,  0.94073996,  0.75232206,  0.39874223,  0.19569684,
         0.07680474,  0.02665377,  0.02955768]]), array([0.03911377])]
>>>>>>>>>>>>>run time:
0.53125s
